lib/routes/slack.js

```diff
@@ -0,0 +1,95 @@
+module.exports = function(crowi, app) {
+  'use strict';
+
+  var slackHelper = require('../util/slack')(crowi)
+  , Page = crowi.model('Page')
+  , User = crowi.model('User')
+  , actions = {};
+
+  var api = actions.api = {};
+
+  function fetchPage(link) {
+  }
+
+  api.handleEvent = function(req, res) {
+    if (req.body.type != null && req.body.type == 'url_verification') {
+      verifyChallenge(req, res);
+      return;
+    }
+
+    let event = req.body.event;
+    switch (event.type) {
+      case 'link_shared':
+      unfurl(req, res);
+      break;
+      default:
+      break;
+    }
+  }
+
+  function verifyChallenge(req, res) {
+    res.send(req.body.challenge);
+  }
+
+  function unfurl(req, res) {
+    const https = require('https');
+    const HOST = 'slack.com'
+    const PATH = '/api/chat.unfurl';
+
+    const link = req.body.event.links[0];
```

Comment:
Related to above comment (of about `%20`), is the link passed by Slack already decoded?